File: server/models/category.js
// Object destructuring, the 'require...' is returning a new model, so,
// we Create another new model and assign all the values of the first one to the new one we created
const { Schema, model } = require('mongoose');

const categorySchema = Schema({
  name: { type: String, required: true },
  colour: { type: String, default: '#000000' },
  image: { type: String, required: true },
  markedForDeletion: { type: Boolean, default: false },
});

categorySchema.set('toJSON', { virtuals: true });
categorySchema.set('toObject', { virtuals: true });

exports.Category = model('Category', categorySchema);

File: Mobile/lib/src/product/domain/entities/category.dart
import 'package:equatable/equatable.dart';

class ProductCategory extends Equatable {
  const ProductCategory({
    required this.id,
    this.name,
    this.colour,
    this.image,
  });

  const ProductCategory.empty() : this(id: "Test String");

  const ProductCategory.all() : this(id: '', name: 'All');

  final String id;
  final String? name;
  final String? colour;
  final String? image;

  @override
  List<dynamic> get props => [
        id,
        name,
        colour,
      ];
}
